lib/demo/accounts/user.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Demo.Accounts.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :email, :string
+    field :password, :string, virtual: true
+    field :hashed_password, :string
+    field :confirmed_at, :naive_datetime
+
+    timestamps()
+  end
+
+  @doc """
+  A user changeset for registration.
+
+  It is important to validate the length of both e-mail and password.
+  Otherwise databases may truncate them without warnings, which could
+  lead to unpredictable or insecure behaviour. Long passwords may also
+  be very expensive to hash.
```

Comment:
```suggestion
  Otherwise databases may truncate the e-mail without warnings, which
  could lead to unpredictable or insecure behaviour. Long passwords may
  also be very expensive to hash for certain algorithms.
```